packages/router/src/grpc/view-protocol-server/note-by-commitment.ts

```diff
@@ -0,0 +1,25 @@
+import {
+  NoteByCommitmentRequest,
+  NoteByCommitmentResponse,
+} from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/view/v1alpha1/view_pb';
+import { ViewReqMessage } from './router';
+import { ServicesInterface } from '@penumbra-zone/types';
+
+export const isNoteByCommitmentRequest = (req: ViewReqMessage): req is NoteByCommitmentRequest => {
+  return req.getType().typeName === NoteByCommitmentRequest.typeName;
+};
+
+export const handleNoteByCommitmentReq = async (
+  req: NoteByCommitmentRequest,
+  services: ServicesInterface,
+): Promise<NoteByCommitmentResponse> => {
+  const { indexedDb } = await services.getWalletServices();
+  if (!req.noteCommitment) return new NoteByCommitmentResponse();
```

Comment:
I feel like we should throw an error. Or does the Rust implementation send an empty response like this?

Reply:
Rust imlementation throws an error. I noticed that in one more request we return an empty response where we should throw an error
https://github.com/penumbra-zone/web/blob/89d4ec182ec5946410bea0b057e45afa00be6552/packages/router/src/grpc/view-protocol-server/tx-info-by-hash.ts#L18